### PR TITLE
FDG-9759 Create New Data Table

### PR DESCRIPTION
**Pull request by nicweidner**
**Jira**
https://federal-spending-transparency.atlassian.net/browse/FDG-9759

**Test Coverage** 
All files: 90.13%
data-preview.tsx: 83.96%
**data-preview-data-table.tsx: 100%**
**data-preview-data-table-body.tsx: 100%**
data-preview-data-table-header.tsx: 73.07%
data-preview-section-container.tsx: 61.83%
data-preview-table.tsx: 75.94%

Unit test coverage is low, but this work will be covered in other tickets: 
data-preview-download.tsx: 68.04%
date-range-filter.tsx: 75.23%
data-preview-pivot-options.tsx: 48.64%

### CODE REVIEW

File: src/components/data-preview/data-preview.tsx [hunk 1]
```diff
@@ -1,14 +1,368 @@
...
+
+    Analytics.event({
+      category: 'Data Table Selector',
+      action: 'Pick Table Click',
+      label: table.tableName,
+    });
+  };
+
+  useEffect(() => {
+    console.log('selectedTable: ', selectedTable);
```
[the remark this answers is not in the document]
Comment (nicweidner) [reply src/components/data-preview/data-preview.tsx:115]: This was used for testing locally and is being removed.

File: src/components/data-preview/data-preview-section-container/data-preview-section-container.tsx [hunk 2]
```diff
@@ -0,0 +1,525 @@
...
+          if (!selectedPivot?.pivotValue) {
+            meta = res.meta;
+            if (totalCount !== 0 && totalCount <= MAX_PAGE_SIZE * 2) {
+              try {
+                return await queryClient.ensureQueryData({
+                  queryKey: ['tableData', selectedTable, from, to, userFilterSelection],
+                  queryFn: () => fetchAllTableData(sortParam, totalCount, selectedTable, apiFilterParam, dateFilter),
+                });
+              } catch (error) {
+                console.warn(error);
```
[the remark this answers is not in the document]
Comment (nicweidner) [reply src/components/data-preview/data-preview-section-container/data-preview-section-container.tsx:166]: False positive.

File: src/components/data-preview/data-preview-section-container/data-preview-section-container.tsx [hunk 3]
```diff
@@ -0,0 +1,525 @@
...
+              setManualPagination(false);
+              return null;
+            }
+          }
+        })
+        .catch(err => {
+          if (err.name === 'AbortError') {
+            console.info('Action cancelled.');
+          } else {
+            console.error('API error', err);
```
[the remark this answers is not in the document]
Comment (nicweidner) [reply src/components/data-preview/data-preview-section-container/data-preview-section-container.tsx:180]: False positive.

File: src/components/data-preview/data-preview-table/data-preview-table.tsx [hunk 4]
```diff
@@ -0,0 +1,471 @@
...
+            if (maxRows !== totalCount) setMaxRows(totalCount);
+            setTableData(res.data);
+          }
+        })
+        .catch(err => {
+          if (startPage === 1) {
+            setRowsShowing({ begin: 0, end: 0 });
+            setMaxRows(0);
+          }
+          console.error(err);
```
[the remark this answers is not in the document]
Comment (nicweidner) [reply src/components/data-preview/data-preview-table/data-preview-table.tsx:234]: False positive.

File: src/components/data-preview/data-preview-section-container/data-preview-section-container.tsx [hunk 5]
```diff
@@ -0,0 +1,431 @@
...
+          if (!selectedPivot?.pivotValue) {
+            meta = res.meta;
+            if (totalCount !== 0 && totalCount <= MAX_PAGE_SIZE * 2) {
+              try {
+                return await queryClient.ensureQueryData({
+                  queryKey: ['tableData', selectedTable, from, to, userFilterSelection],
+                  queryFn: () => fetchAllTableData(sortParam, totalCount, selectedTable, apiFilterParam, dateFilter),
+                });
+              } catch (error) {
+                console.warn(error);
```
[the remark this answers is not in the document]
Comment (nicweidner) [reply src/components/data-preview/data-preview-section-container/data-preview-section-container.tsx:160]: False positive

File: src/components/data-preview/data-preview-section-container/data-preview-section-container.tsx [hunk 6]
```diff
@@ -0,0 +1,431 @@
...
+              setManualPagination(false);
+              return null;
+            }
+          }
+        })
+        .catch(err => {
+          if (err.name === 'AbortError') {
+            console.info('Action cancelled.');
+          } else {
+            console.error('API error', err);
```
[the remark this answers is not in the document]
Comment (nicweidner) [reply src/components/data-preview/data-preview-section-container/data-preview-section-container.tsx:174]: False positive

**Review by wyattch — APPROVED** [review]
Awesome work! Approved.

**Pull request merged**